sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchServiceResourceClientBuilder.java

```diff
@@ -102,40 +102,40 @@ public SearchServiceResourceClientBuilder() {
     }
 
     /**
-     * Creates a {@link SearchDataSourceClient} based on options set in the Builder. Every time
-     * {@code buildDataSourceClient()} is called a new instance of {@link SearchDataSourceClient} is created.
+     * Creates a {@link SearchIndexerDataSourceClient} based on options set in the Builder. Every time
+     * {@code buildDataSourceClient()} is called a new instance of {@link SearchIndexerDataSourceClient} is created.
      * <p>
      * If {@link #pipeline(HttpPipeline) pipeline} is set, then only the {@code pipeline},
      * {@link #serviceVersion(SearchServiceVersion)} and {@link #endpoint(String) endpoint}
-     * are used to create the {@link SearchDataSourceClient client}. All other builder settings are ignored.
+     * are used to create the {@link SearchIndexerDataSourceClient client}. All other builder settings are ignored.
      * </p>
      * @return A SearchDataSourceClient with the options set from the builder.
      * @throws NullPointerException If {@code endpoint} are {@code null}.
      */
-    public SearchDataSourceClient buildDataSourceClient() {
-        return new SearchDataSourceClient(buildDataSourceAsyncClient());
+    public SearchIndexerDataSourceClient buildDataSourceClient() {
```

Comment:
I thought we agreed these wouldn't be instantiatable/buildable by themselves, but can only be obtained from `SearchServiceClient`. These are not full clients but only used for "grouping" methods. I have a meeting with @tg-msft and @KrzysztofCwalina on Monday to solidify this, but was otherwise the POR.

Reply:
@alzimmermsft 
I tried out to have one entry point. But Java has some limitation of initializing subclients under different packages without using public constructors/builders.

Reply:
This is the same issue we saw with Storage where Java doesn't have a module/DLL internal scoping so we either need to make the constructor public or offer a builder that would be able to create this. In Storage we went with the latter.